tests/test_domain_objects/shared_tests.py

```diff
@@ -0,0 +1,197 @@
+
+
+def isin_exists(record):
+    isins = helper.query_db('instruments', 'isin')
+    assert record['isin'] in isins
+
+
+def isin_valid(record):
+    """ Valid ISIN is in format '"COI""CUSIP""4"' """
+    coi = record['coi']
+    cusip = str(record['cusip'])
+    expected_isin = coi + cusip + '4'
+    assert record['isin'] == expected_isin
+
+
+def ric_exists(record):
+    rics = helper.query_db('instruments', 'ric')
+    ric = record['ric']
+    assert ric in rics
+
+
+def ric_valid(record):
+    """ RIC Formatting should be a ticker followed by an exchange code.
+    In this use case, exchange codes are currently integers. This test
+    will require updating once this generation approach changes to real-world
+    exchange codes. """
+    ric = record['ric']
+    ticker = record['ticker']
+    id = str(record['instrument_id'])
+    assert valid_ric(ticker, ric, id)
+
+
+def valid_ric(ticker, ric, id):
+    """ Valid RIC is in format '"ticker"."id"' """
+    split_ric = ric.split('.')
+    ticker_ = split_ric[0]
+    id_ = split_ric[1]
+    return ticker == ticker_ and id == id_
+
+
+def unique_ids(records, object_name):
+    id_set = set()
+    id_field_name = object_name+'_id'
+
+    for record in records:
+            id_set.add(record[id_field_name])
+    assert len(records) == len(id_set)
+
+
+def dummy_fields_valid(record, object_name):
+    partial_field_name = object_name+'_field'
+    dummy_field_names = get_dummy_field_names(record, partial_field_name)
+
+    for dummy_field in dummy_field_names:
+        if not is_length(10, record[dummy_field]):
+            assert False
+    assert True
+
+
+def get_dummy_field_names(record, partial_field_name):
+    dummy_field_names = []
+    for column_name in record.keys():
+        if partial_field_name in column_name:
+            dummy_field_names.append(column_name)
+    return dummy_field_names
+
+
+def long_short_valid(record):
+    long_short_values = domain_obj.LONG_SHORT
+    long_short_value = record['long_short']
+    assert long_short_value in long_short_values
+
+
+def purpose_valid(record, purposes):
+    purpose = record['purpose']
+    assert purpose in purposes
+
+
+def account_number_valid(record):
+    account_number = record['account_num']
+    assert is_int(account_number) and is_length(8, account_number)
+
+
+def currency_valid(record):
+    cache = Cache()
+    currencies = cache.retrieve_from_cache('currencies')
+    currency = record['currency']
+    assert currency in currencies
+
+
+def amount_valid(record):
+    amount = record['amount']
+    assert 1 <= amount <= 10000
+
+
+def quantity_valid(record, long_short='Long'):
+    quantity = record['qty']
+    if (long_short == 'Long'):
+        assert 1 <= quantity <= 10000
+    else:
+        assert -10000 <= quantity <= -1
+
+
+def direction_valid(record):
+    assert record['direction'] in domain_obj.CREDIT_DEBIT
+
+
+def account_valid(record, account_types):
+    assert record['account'][:3] in account_types
```

Comment:
check account Id is a valid format (integer castabale?). 
check the account exists in the DB
check the account type is valid (DEPOT for a depot position, ICP for back and front office and ECP for the counterparty [pending question responses])

Reply:
ISSUE

Reply:
https://github.com/galatea-associates/fuse-test-data-gen/issues/50